crates/shared/src/signature_validator/simulation.rs

```diff
@@ -18,18 +19,54 @@ pub struct Validator {
     signatures: contracts::support::Signatures,
     settlement: H160,
     vault_relayer: H160,
+    web3: Web3,
 }
 
 impl Validator {
+    /// The result returned from `isValidSignature` if the signature is correct
+    const IS_VALID_SIGNATURE_MAGIC_BYTES: &'static str = "1626ba7e";
+
     pub fn new(web3: &Web3, settlement: H160, vault_relayer: H160) -> Self {
         let web3 = ethrpc::instrumented::instrument_with_label(web3, "signatureValidation".into());
         Self {
             signatures: contracts::support::Signatures::at(&web3, settlement),
             settlement,
             vault_relayer,
+            web3: web3.clone(),
         }
     }
 
+    /// Simulate isValidSignature for the cases in which the order does not have
+    /// pre-interactions
+    async fn simulate_without_pre_interactions(
+        &self,
+        check: &SignatureCheck,
+    ) -> Result<Simulation, SignatureValidationError> {
+        // We simulate the signature verification from the owner contract's
+        // context. This allows us to simulate the actual `isValidSignature` calls that
+        // would happen as part of a settlement
+        let contract = ERC1271SignatureValidator::at(&self.web3, check.signer);
+        let magic_bytes = contract
+            .methods()
+            .is_valid_signature(Bytes(check.hash), Bytes(check.signature.clone()))
+            .call()
+            .await
+            .map(|value| hex::encode(value.0))
+            .map_err(|e| {
+                tracing::error!(
```

Comment:
This could be a little bit noisy for the oncall rotation. But I wanted to really flag this error, because I tested it for gnosis chain, and in theory all owners with EIP-1271 must implement the `isValidSignature` method.

Reply:
> This could be a little bit noisy for the oncall rotation

If we have an order with this error we get an alert every time we build an auction. That's not acceptable. For the purposes of this code it doesn't really matter if `isValidSignature()` returned a wrong value or the contract doesn't have the function at all. Ultimately the order is not usable so it just needs to be filtered out.

Reply:
@MartinquaXD it would be ok if I set it as `warn`? I want something queryable I can use in elasticsearch to see how many of these cases we have.

Reply:
There is no need to introduce this IMO. The current logic already uses `SignatureValidationError::Invalid` when the call reverts which also happens when the function does not exist. Since this didn't cause any issues so far I don't see why the simpler check should have different semantics than the original one.